lab_3/main.py

```diff
@@ -222,7 +331,24 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        profile = {}
+        freq_dict = {}
+        new_key = ''
+        with open(name, 'w', encoding="utf-8") as file:
+            for trie in self.tries:
+                for key, value in trie.n_gram_frequencies.items():
+                    for letter_id in key:
+                        new_key += self.storage.get_letter_by_id(letter_id)
+                    freq_dict[new_key] = value
+                    new_key = ''
+            profile['freq'] = freq_dict
+            profile['n_words'] = self.n_words
+            profile['name'] = self.language
+            json_string = json.dumps(profile)
+            file.write(json_string)
```

Comment:
There is the `.dump` method in the `json` package which combines these two lines.

Reply:
done